### ISSUE TITLE
Get amp handler in more decent way

### ISSUE THREAD

**Issue by matthew-z** [NVIDIA/apex#98]
Usually, the location of `amp.init()` is far from `loss.backward()`. While it is possible to pass the handler as a parameter to the function which calls `loss.backward()`, but it is not very decent. I wonder if we can do something like:

```
import torch
import apex
apex.amp.apex.init()

........


def backward(loss):
    with apex.amp.get_default_handler().scale_loss(loss, optimizer) as scaled_loss:
        scaled_loss.backward()
    optimizer.step()
```
